src/ert/config/observations.py

```diff
@@ -613,4 +613,11 @@ def from_ert_config(  # pylint: disable=too-many-branches
         return EnkfObs({}, obs_time_list)
 
     def get_dataset(self, key: str) -> Tuple[str, xr.Dataset]:
-        return self[key].to_dataset(self, [])
+        response_key, dataset = self[key].to_dataset(self, [])
+        dataset.attrs["response"] = response_key
```

Comment:
We should unify the API here and make the response key be always part of the attributes of the dataset

Reply:
I put it in the a new commit, but can merge it with the first one if you want